sdk/core/System.ClientModel/api/System.ClientModel.net6.0.cs

```diff
@@ -1,43 +1,40 @@
-        public static T? Read<T>(System.BinaryData data, System.ClientModel.ModelReaderWriterOptions? options = null) where T : System.ClientModel.Primitives.IPersistableModel<T> { throw null; }
-        public static System.BinaryData Write(object model, System.ClientModel.ModelReaderWriterOptions? options = null) { throw null; }
-        public static System.BinaryData Write<T>(T model, System.ClientModel.ModelReaderWriterOptions? options = null) where T : System.ClientModel.Primitives.IPersistableModel<T> { throw null; }
-    }
-    public partial class ModelReaderWriterOptions
-    {
-        public ModelReaderWriterOptions(string format) { }
-        public string Format { get { throw null; } }
-        public static System.ClientModel.ModelReaderWriterOptions Json { get { throw null; } }
-        public static System.ClientModel.ModelReaderWriterOptions Xml { get { throw null; } }
-    }
-}
 namespace System.ClientModel.Primitives
 {
     public partial interface IJsonModel<out T> : System.ClientModel.Primitives.IPersistableModel<T>
     {
-        T Create(ref System.Text.Json.Utf8JsonReader reader, System.ClientModel.ModelReaderWriterOptions options);
-        void Write(System.Text.Json.Utf8JsonWriter writer, System.ClientModel.ModelReaderWriterOptions options);
+        T Create(ref System.Text.Json.Utf8JsonReader reader, System.ClientModel.Primitives.ModelReaderWriterOptions options);
+        void Write(System.Text.Json.Utf8JsonWriter writer, System.ClientModel.Primitives.ModelReaderWriterOptions options);
     }
     public partial interface IPersistableModel<out T>
     {
-        T Create(System.BinaryData data, System.ClientModel.ModelReaderWriterOptions options);
-        string GetFormatFromOptions(System.ClientModel.ModelReaderWriterOptions options);
-        System.BinaryData Write(System.ClientModel.ModelReaderWriterOptions options);
+        T Create(System.BinaryData data, System.ClientModel.Primitives.ModelReaderWriterOptions options);
+        string GetFormatFromOptions(System.ClientModel.Primitives.ModelReaderWriterOptions options);
+        System.BinaryData Write(System.ClientModel.Primitives.ModelReaderWriterOptions options);
     }
     [System.Diagnostics.CodeAnalysis.RequiresUnreferencedCodeAttribute("The constructors of the type being deserialized are dynamically accessed and may be trimmed.")]
-    public partial class ModelJsonConverter : System.Text.Json.Serialization.JsonConverter<System.ClientModel.Primitives.IJsonModel<object>>
+    public partial class JsonModelConverter : System.Text.Json.Serialization.JsonConverter<System.ClientModel.Primitives.IJsonModel<object>>
```

Comment:
Why are we associating the converter with `IJsonModel` which we don't think anyone should ever be concerned with instead of `ModelReaderWriter` that's the champion scenario of this feature area?

Reply:
The converter works without ModelReaderWriter, i.e. has close to nothing to do with the ModelReaderWriter. All you need to use it is JsonSerializer and a model that implements IJsonModel.

Reply:
Because this class only converts IJsonModel.  ModelReaderWriter works with more than just that.

Reply:
>  has close to nothing to do with the ModelReaderWriter

99.999% of `ModelReaderWriter`'s implementation is calling `IJsonModel` to do all the heavy lifting.  I think this converter has a lot to do with `ModelReaderWriter` at the conceptual level.  It even takes the exact same `ModelReaderWriterOptions` class we pass to `ModelReaderWriter`.  We're expecting `IJsonModel` will mean nothing to most folks and `ModelReaderWriter` will be the API they discover/use.  I'd rather name this after the way people should think about using it instead of an implementation detail we don't expect them to know or care about.

Reply:
I don't follow:

> 99.999% of ModelReaderWriter's implementation is calling IJsonModel to do all the heavy lifting. 

> We're expecting IJsonModel will mean nothing to most folks and ModelReaderWriter will be the API they discover/use. 

Both statements are true, but what do they have to do with JsonModelConverter?  

> I'd rather name this after the way people should think about using it instead of an implementation detail we don't expect them to know or care about.

I feel like your proposed name is based on implementation detail. The fact that ModelReaderWriter used JsonSerializer is an implementation detail. 

It's possible that you are not aware how this converter is meant to be used. It's meant to be passed to JsonSerializer, not applied to model types.

Reply:
> what do they have to do with JsonModelConverter?

The converter only exists as a convenience API, like `ModelReaderWriter`.  It's the easy way to use `ModelReaderWriter` to serialize Azure types that are members of your own objects with `JsonSerializer`.

> I feel like your proposed name is based on implementation detail. The fact that ModelReaderWriter uses JsonSerializer is an implementation detail.

Now I don't follow.  :smile:  `ModelReaderWriter` doesn't use `JsonSerializer`.

> It's possible that you are not aware how this converter is meant to be used. It's meant to be passed to JsonSerializer, not applied to model types.

It's meant for customers to use when they want one of our Azure types as a member of their object.  It's just a convenience to integrate `ModelReaderWriter` with `JsonSerializer` for customer convenience.  We'll never use this internally.  Am I wrong about any of that?

I think the way people approach this converter will be "how do I use `ModelReaderWriter` with `JsonSerializer` when I'm putting Azure models in my custom type."  If that's the case, it feels like our internal `IJsonModel` interface that we added to make `ModelReaderWriter` more performant is an unhelpful implementation detail to use for the name here.

Reply:
I also thing it is quite common to use `[ClassName]Converter` when inheriting from `JsonConverter`.  This class does not convert `ModelReaderWriter's` it converts anything that implements `IJsonModel<T>`.

Reply:
Its the convention we follow everywhere else https://grep.app/search?q=%3A%20JsonConverter&filter[repo][0]=Azure/azure-sdk-for-net

Reply:
ModelReaderWriter does NOT use `JsonSerializer` ever just for clarification above.  JsonSerializer uses ModelReaderWriter when the JsonModelConverter is added to its options.

Reply:
Talked offline and we agreed to make this internal for now.  I have opened up this issue to track this but it will not be required for the initial GA.

https://github.com/Azure/azure-sdk-for-net/issues/40778